src/App.tsx

```diff
@@ -1,56 +1,60 @@
-import React from 'react';
 import './App.scss';
+import classNames from 'classnames';
+import { debounce } from 'lodash';
+import { useState, useCallback, useMemo } from 'react';
+import { PeopleList } from './components/peopleList';
+
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [inputValue, setInputValue] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isListVisible, setIsListVisible] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const delay = 1000;
+
+  const aplyQuery = useCallback(debounce(setAppliedQuery, delay), []);
+
+  const onChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+    aplyQuery(event.target.value);
+  };
+
+  const filteredPeople = useMemo(() => (
+    [...peopleFromServer]
+      .filter((person) => (
+        person.name.toLowerCase().includes(appliedQuery.toLowerCase())
+      ))
+  ), [peopleFromServer, appliedQuery]);
+
+  const selectHandler = (person: Person) => {
+    setSelectedPerson(person);
+    setAppliedQuery('');
+    setInputValue('');
```

Comment:
Do not clear input value after selecting a person, save selected suggestion text to the input